### DISCUSSION TITLE
Comprehensive list of blockers/dealbreakers as to why UTM can't be put on the iOS/iPadOS App store

### DISCUSSION THREAD

**Discussion by ShadowJonathan** [utmapp/UTM#3136] [UTM Help (iOS)] [answered]
UTM looks great, however, when reading through the FAQ and the install process requirements, I couldn't find a good reason as to why UTM hasn't "simply" been submitted to the App Store.

What exactly are the problems that come up with publishing this app to the store? The requirement to not have JIT in the app? Some other app requirement? Does apple's store policies simply not allow virtualisation apps?

**Comment by osy** [comment, marked as answer]
1. Apple expressly prohibits JIT AND interpreted code. (There’s exceptions but keep reading.)
2. Apple can reject your app for any reason. It’s within their rights. If they don’t like the “spirit” of your app they’ll reject it.
3. Why do you say it hasn’t been submitted? What makes you think it hasn’t been tried?
4. Historically devs have claimed that Apple made their lives harder when they publicize conversations they have with Apple such that other things they submit would take longer time and get higher scrutiny.
